src/charm.py

```diff
@@ -419,6 +424,46 @@ def _on_upgrade_charm(self, _event: UpgradeCharmEvent) -> None:
         runner_manager.flush()
         self._reconcile_runners(runner_manager)
 
+    def _configure_aproxy(self, aproxy_proxy: str) -> None:
+        """Configure aproxy.
+
+        Args:
+            aproxy_proxy: Proxy to configure aproxy with.
+        """
+        logger.info("Configuring aproxy for the runner.")
+
+        execute_command(["snap", "set", "aproxy", f"proxy={aproxy_proxy}"])
+
+        stdout, _ = execute_command(
+            [
+                "/bin/bash",
+                "-c",
+                r"ip route get $(ip route show 0.0.0.0/0 | grep -oP 'via \K\S+') |"
+                r" grep -oP 'src \K\S+'",
+            ],
+            check_exit=True,
+        )
+        default_ip = stdout.strip()
+
+        nft_input = f"define default-ip = {default_ip}"
+        nft_input += r"""
```

Comment:
```suggestion
        nft_input = f"""define default-ip = {default_ip}"
```

f-string and triple quotes can be used together.

Reply:
Yes, I know, but my intention  was to avoid f raw strings. But I have updated the code now.

Reply:
Yes, indeed, if we have to use the f string, we need to escape all the curly braces. Sorry I missed that, up to you to use your previous approach or use the f string.